### PR TITLE
[nettle] Switch from yasm to clang

### PR DESCRIPTION
**Pull request by dg0yt**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by Neumann-A** [comment]
gmp has ASMFLAGS for that instead of cflags

**Comment by dg0yt** [comment]
Building successfully now. Anyone out there who can validate the usability of the binary artifacts?
For arm-uwp and arm64-windows, there are no consuming ports. (shiftmedia-libgnutls doesn't support arm.)

File: ports/gmp/portfile.cmake [hunk 1]
```diff
@@ -43,6 +43,10 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     )
 endif()
 
+if(VCPKG_TARGET_IS_ANDROID AND VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    vcpkg_list(APPEND OPTIONS "CFLAGS=\$CFLAGS -m32")
```
Comment (dg0yt) [inline ports/gmp/portfile.cmake:47]: Found during x86 tests of nettle with the NDK's clang on linux host. To be removed from this PR.

File: ports/nettle/portfile.cmake [hunk 2]
```diff
@@ -10,37 +9,73 @@ vcpkg_from_gitlab(
...
+        #nettle_cv_asm_type_percent_function=no
+        #nettle_cv_asm_align_log=no
     )
+    if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+        string(APPEND asmflags " --target=i686-pc-windows-msvc -m32")
+    elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+        string(APPEND asmflags " --target=x86_64-pc-windows-msvc")
+    elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
+        string(APPEND asmflags " --target=arm64-pc-windows-msvc")
+    else()
```
Comment (dg0yt) [inline ports/nettle/portfile.cmake:61]: Oh, no asm for arm-uwp

File: ports/gmp/portfile.cmake (hunk 1 above)
Comment (Neumann-A) [reply ports/gmp/portfile.cmake:47]: That flag has to come from the toolchain.

Comment (dg0yt) [reply ports/gmp/portfile.cmake:47]: But from which point in the toolchain setup?
I took the `-m32` from https://developer.android.com/ndk/guides/abis#x86, but I didn't find it in the NDK's or CMake's Android toolchain initialization.

Comment (dg0yt) [reply ports/gmp/portfile.cmake]: ... and I also had to add `-m32` to nettle's ASMFLAGS for clang-cl x86

**Comment by MonicaLiu0311** [comment]
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
x64-linux
x64-osx
arm-uwp
arm64-windows
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**